adapters/nsw.js

```diff
@@ -0,0 +1,158 @@
+    } else if (idy === 1) {
+      colCounter = 0;
+
+      $(this).children().each(function (idx, cell) {
+        // The column counter needs to take occasional colspans into account
+        var colIncrement = Number($(this).attr('colspan')) || 1;
+        colCounter += colIncrement;
+
+        var regExp = new RegExp('([0-9]+)-hour');
+        var period = $(this).html().match(regExp);
+        if (period) {
+          indexParams[_.findIndex(indexParams, { 'col': colCounter })].avgPeriod = period[1];
+        }
+      });
+
+    // Any other row may contain measurements
+    } else {
+      colCounter = 0;
+
+      // The existence of <td class='site'> is an indication of a measurement
+      if ($(this).find('.site').length) {
+        // Each region may have several sites (rows), but is only mentioned
+        // once: in the row of its first site.
+        var regionString = $('.region', this).text();
+        region = regionString || region;
+
+        var site = $('.site', this).text();
+
+        // Store the main properties for this measuring station
+        var base = {
+          city: region,
+          location: site,
+          date: date,
+          attribution: [{
+            name: 'NSW - Office of Environment & Heritage',
+            url: 'http://www.environment.nsw.gov.au/'
+          }]
+        };
+
+        $(this).children().each(function (idx, cell) {
+          // The column counter needs to take occasional colspans into account
+          var colIncrement = Number($(this).attr('colspan')) || 1;
+          colCounter += colIncrement;
+
+          // Check if there is a value being reported and if the pollutant
+          // is available in the indexParams
+          var i = _.findIndex(indexParams, 'col', colCounter);
+          if ($(this).text() && i !== -1) {
+            var p = indexParams[i];
+
+            var m = _.clone(base);
+            m.parameter = p.parameter;
+            m.unit = units[p.parameter];
+            m.value = Number($(this).text());
+            m.averagingPeriod = {value: Number(p.avgPeriod), unit: 'hours'};
+            measurements.push(m);
+          }
+        });
+      }
+    }
+  });
+
+  // The same pollutant can contain multiple measurements per location.
+  // In this case, keep the measurement with the SHORTEST averaging period.
+  var finalMeasurements = [];
+  for (var m in measurements) {
+    // Check if this location already has a measurement for this parameter
+    var match = _.findIndex(finalMeasurements, _.matches({'parameter': measurements[m].parameter, 'location': measurements[m].location}));
+    if (match === -1) {
+      // No matching measurement found
+      finalMeasurements.push(measurements[m]);
+    } else if (measurements[m].averagingPeriod.value < finalMeasurements[match].averagingPeriod.value) {
+      // Matching measurement found with bigger avgPeriod
+      finalMeasurements.push(measurements[m]);
+      finalMeasurements.splice(match, 1);
```

Comment:
If you wanted, you could probably do the loop in reverse order and then just delete the item instead of splicing it, but this seems to work equally as well.